clang/lib/Driver/Driver.cpp

```diff
@@ -733,8 +733,23 @@ void Driver::CreateOffloadingDeviceToolChains(Compilation &C,
           C.getInputArgs().getLastArg(options::OPT_fsycl_link_targets_EQ);
   Arg *SYCLAddTargets =
           C.getInputArgs().getLastArg(options::OPT_fsycl_add_targets_EQ);
+  bool SomeSYCLTargetsOption = SYCLTargets || SYCLLinkTargets || SYCLAddTargets;
   bool SYCLLink = C.getInputArgs().hasArg(options::OPT_fsycl_link_EQ);
   bool SYCLfpga = C.getInputArgs().hasArg(options::OPT_fintelfpga);
+
+  // Start SYCL options conjunction checks
+  // -fsycl*targets option must come with -fsycl
+  if (SomeSYCLTargetsOption && !HasValidSYCLRuntime) {
+    std::string SYCLTargetsType;
+    // Checks priority: -fsycl-targets, -fsycl-link-targets,
+    // -fsycl-add-targets
+    SYCLTargetsType = SYCLTargets ? "-" : "-link-";
+    if (SYCLTargetsType.empty())
+      SYCLTargetsType = "-add-";
```

Comment:
This will diagnose **-link-** both for -link- and -add-. This needs to be fixed and, alas, covered with extra testing.